### ISSUE TITLE
pyls not using flake8 but pylint

### ISSUE THREAD

**Issue by sohnryang** [palantir/python-language-server#554]
I'm using pyls for neovim with [autozimu/LanguageClient-neovim](https://github.com/autozimu/LanguageClient-neovim). I want to use flake8 instead of pylint since pylint has performance issues with my project, while flake8 is perfectly fine. I set `.vim/settings.json` like this (in the project root):

```json
{
    "pyls": {
        "configurationSources": ["flake8"]
    }
}
```

Also, to set flake8, I set `setup.cfg` like this:

```ini
[flake8]
max-line-length = 80
```

However, pyls is ignoring my configuration and it is still using pyflakes. I want to know what to change in my config to use flake8 instead of pylint. Thanks in advance.

**Comment by sohnryang** [comment]
In case if you think the problem is in my project setup, you can take a look at my python project [here](https://github.com/sohnryang/buffon-needle-simulator).

**Comment by mgorfer** [comment]
I just started using pyls recently and had the same problem. But I think, I just figured it out. I am using pyls with spacemacs, so the code is a bit different, but I think the result should be the same.

The `lsp-pyls-configuration-sources` is not responsible for choosing the checking tool itself (flake8, pycodestyle or pylint). It just chooses which config it should use for all of them. So if you use the standard setting  `lsp-pyls-configuration-sources: ["pycodestyle"]` and your setup.cfg contains
```
[pycodestyle]
max-line-length = 100
```
then pycodestyle itself but also flake8 and pylint will not warn if your lines are longer than 79.

Because pylint is also really slowing down my spacemacs I disabled it with `lsp-pyls-plugins-pylint-enable: nil`
